hedera-node/src/main/java/com/hedera/services/txns/token/TokenCreateTransitionLogic.java

```diff
@@ -62,137 +72,150 @@
+			ledger.updateTokenXfers(provisionalTokenId, treasuryGrpcId, op.getInitialSupply());
+		}
+		persistableRelationships.add(treasuryRel);
+
+		/* --- Persist anything modified/new --- */
+		for (final var account : persistableAccounts) {
+			accountStore.persistAccount(account);
+			ledger.setAssociatedTokens(account.getId().asGrpcAccount(), new MerkleAccountTokens(account.getAssociatedTokens()));
 		}
+		typedTokenStore.persistTokenRelationships(persistableRelationships);
+		typedTokenStore.persistNew(created);
 
-		store.commitCreation();
-		txnCtx.setCreated(created);
-		txnCtx.setStatus(SUCCESS);
+		/* --- Maintain backwards compatibility for TokenDelete op --- */
+		store.addKnownTreasury(treasuryGrpcId, provisionalTokenId);
 	}
 
-	private ResponseCodeEnum autoEnableFeeCollectors(
-			TokenID created,
+	private void validateCreation(TokenCreateTransactionBody op) {
+		validateFalse(op.hasExpiry() && !validator.isValidExpiry(op.getExpiry()), INVALID_EXPIRATION_TIME);
+	}
+
+	private void autoEnableFeeCollectors(
+			Token created,
 			AccountID alreadyEnabledTreasury,
-			TokenCreateTransactionBody op
+			TokenCreateTransactionBody op,
+			List<Account> persistableAccounts,
+			List<TokenRelationship> persistableRelationships
 	) {
-		/* TokenStore.associate() returns TOKEN_ALREADY_ASSOCIATED_TO_ACCOUNT if the
-		account-token relationship already exists. */
 		final Set<AccountID> customCollectorsEnabled = new HashSet<>();
 		customCollectorsEnabled.add(alreadyEnabledTreasury);
-		ResponseCodeEnum status = OK;
 		for (var fee : op.getCustomFeesList()) {
 			boolean collectorEnablementNeeded = fee.hasFractionalFee();
 			if (fee.hasFixedFee() && fee.getFixedFee().hasDenominatingTokenId()) {
 				collectorEnablementNeeded |= (0 == fee.getFixedFee().getDenominatingTokenId().getTokenNum());
 			}
 			final var collector = fee.getFeeCollectorAccountId();
+			final var collectorModel = accountStore.loadAccount(Id.fromGrpcAccount(collector));
 			if (collectorEnablementNeeded && !customCollectorsEnabled.contains(collector)) {
-				status = autoEnableAccountForNewToken(collector, created, op);
-				if (status != OK) {
-					return status;
-				}
+				final var rel = autoEnableAccountForNewToken(collectorModel, created, persistableAccounts);
+				persistableRelationships.add(rel);
 				customCollectorsEnabled.add(collector);
 			}
 		}
-		return status;
 	}
 
-	private ResponseCodeEnum autoEnableAccountForNewToken(
-			AccountID id,
-			TokenID created,
-			TokenCreateTransactionBody op
+	private TokenRelationship autoEnableAccountForNewToken(
+			Account account,
+			Token token,
+			List<Account> persistableAccounts
 	) {
-		var status = store.associate(id, List.of(created));
-		if (status != OK) {
-			return status;
-		}
-		if (op.hasFreezeKey()) {
-			status = ledger.unfreeze(id, created);
-		}
-		if (status == OK && op.hasKycKey()) {
-			status = ledger.grantKyc(id, created);
+		TokenRelationship rel = null;
```

Comment:
Instead of shifting again the logic from the Model objects to the transition logic, we have to be mindful why we started the refactoring. 

I think that it would be better to modify `token.newRelationshipWith(account);` method or introduce another one that creates the relationship and sets the correct `setKycGranted` or `setFrozen` and in the transition logic to have:

```Java
TokenRelationship rel = token.newRelationshipWith(account);
if (!account.getAssociatedTokens().contains(token.getId())) {
	account.associateWith(List.of(token), dynamicProperties.maxTokensPerAccount());
}
```


p.s there is no need for try/catches etc..